### ISSUE TITLE
[fleet] Add `apm-server.url` to the APM fleet configuration 

### ISSUE THREAD

**Issue by simitt** [elastic/apm-server#4895]
When running on cloud or via ECK the publicly reachable APM Server `URL` differs from the server's `host`. 
Since the public URL is not visible anywhere else, we need to ensure it shows up in the APM Server integration in the Fleet UI. 

~The APM Setup Guide currently embeds the APM Server URL + secret token for user convenience when setting up APM agents and will have to retrieve this information from the agent policy (see https://github.com/elastic/kibana/issues/89989).~

The public URL can be injected when the managed policy is created. It doesn't have any direct impact on the APM Server itself.

**Comment by axw** [comment]
@simitt instead of adding another var to the policy template, could we have ESS/ECK/etc. add a "config" value (as opposed to a var defined in the policy template)?

In my prototype branch for injecting agent config into policies, there's an example of adding arbitrary config:

https://github.com/elastic/kibana/blob/6205fbe62eb7e0bfcbb2ef48e8745c538c2b7aee/x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts#L370

**Comment by simitt** [comment]
Good idea - I believe this should be possible.

**Comment by axw** [comment]
@simitt if we can do that then there will be no changes required in apm-server or the integration package. The config will effectively be a conduit between ESS and code in Kibana.

Can we close this and reopen if needed?

**Comment by simitt** [comment]
I updated the description accordingly. We downscaled the efforts around the Getting Started guide, which will not show the APM Server URL + secret token for APM Server integrations, but only link to the Fleet UI. 

Config values do not show up in the UI (afaik), therefore I keep my proposal to add a `URL` config option. It should default to the `hosts` value, will be injected in cloud, and can be set for on-prem installations. It only has an informal purpose.
